cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaConfigurationDiff.java

```diff
@@ -0,0 +1,210 @@
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.replace(" ", "").startsWith("#") && !line.isEmpty())   {
+                validLines.add(line.replace(" ", ""));
+            }
+        }
+
+        return validLines;
+    }
+
+    public Set<String> getDiff() {
+        Properties desiredConfig = configMap2Properties();
+        Set<String> result = new HashSet<>();
+        Map.Entry<ConfigResource, Config> currentBrokerConfiguration = current.entrySet().iterator().next();
+
+        log.info("Diff for broker id {}", currentBrokerConfiguration.getKey());
+        currentBrokerConfiguration.getValue().entries().forEach(entry -> {
+            String val = entry.value() == null ? "null" : entry.value();
+
+            if (desiredConfig.containsKey(entry.name())) {
+                if (desiredConfig.getProperty(entry.name()).equals(val)) {
+                    // entry is in both, equal
+                } else {
+                    // entry is in both, not equal -> diff
+                    if (desiredConfig.getProperty(entry.name()) != null) {
+                        // diff is in pod dependent entry
+                        if (!desiredConfig.getProperty(entry.name()).matches(".*\\$\\{.+\\}.*")) {
+                            result.add(entry.name());
+                            log.debug("diff {} {}/{}", entry.name(), desiredConfig.getProperty(entry.name()), entry.value());
+                        } else {
+                            log.debug("skipping pod-dependent entry {} with value {}", entry.name(), desiredConfig.getProperty(entry.name()));
+                        }
+                    } else {
+                        result.add(entry.name());
+                        log.debug("diff {} {}/{}", entry.name(), desiredConfig.getProperty(entry.name()), entry.value());
+                    }
+                }
+            } else {
+                // desired does not contain key, check default value
+                if (entry.isDefault()) {
+                    // value is default, do nothing
+                } else {
+                    // value is not default, not present in desired -> set it to the default
+                    log.debug("diff {} {}/{}", entry.name(), desiredConfig.getProperty(entry.name()), entry.value());
+                    result.add(entry.name());
+                }
+            }
+        });
+
+        // now we have to check entries added by user
```

Comment:
How does this know from the file if it is added by user or not from these two configurations? By added by the user I expect `Kafka.spec.kafka.config`. Does it mean something else here? If yes, it is confusing.